### DISCUSSION TITLE
Refresh Cognito credentials after 1 hour expiration

### DISCUSSION THREAD

**Discussion by pauloargenal** [aws/aws-sdk-js#4084] [General]
Confirm by changing [ ] to [x] below:
- [x] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js) for answers

Hello! I have a concerned so how do you automatically refresh you AWS SDK session?  I tried different things  using `.refresh()`. and It seems its not automatically updating or even resetting my `AWS.config.credentials`?  What am i doing wrong here?
my code below:
 ```Javascript
import * as AWS from 'aws-sdk'
const handleAWSCredentials = (config) => {
  AWS.config.region = config.region
  AWS.config.credentials = new AWS.CognitoIdentityCredentials({
    IdentityId: config.identityId,
    Logins: {
      [config.identityProviderName]: config.identityToken
    }
  })
}

const handleAWSSession = (setUploadConfig) => {
  const state = localStorage.getItem('state')

  const { config, auth } = JSON.parse(state)

  if(!config.region) {
    uploadConfiguration(setUploadConfig, auth)
    .then((uploadConfiguration) => {
      handleAWSCredentials(uploadConfiguration)
    })
  } else {
    handleAWSCredentials(config)
  }

  AWS.config.credentials.get(err => {
    if(err || AWS.config.credentials.expired) {
      uploadConfiguration(setUploadConfig, auth)
      .then(async (uploadConfiguration) => {
        const { identityProviderName, identityToken } = uploadConfiguration 
        AWS.config.credentials.params.Logins[identityProviderName] = identityToken
        await AWS.config.credentials.getPromise()
      })
    } 
  });
  return config
}
```

**Comment by ajredniwja** [comment]
Hey @Selenophilia thanks for opening this issue, I am not quite able to understand the question here. Are your credentials expiring before time? Can you share how you are using these credentials and what your use case is?

**Comment by pauloargenal** [comment]
Hi @ajredniwja,
 
 Thanks for your response! No,  Our credentials is expiring after 1 hour . I just notice this because I kept seeing  this error on our logs `NotAuthorizedException Invalid login token. Token expired: 1622382209 >= 1622315943 `. Upon searching this error is about the expiration of `aws credentials`. I tried doing some work arounds like storing our aws credentials in a local storage then retrieving and updating them when it expires but doing so is we are experiencing new error namely ` Missing required key 'IdentityPoolId' in params` . I am wondering what am I doing wrong here I tried using the following methods  ( .get() , .refresh(), and .getPromise()) to refresh this credentials and manually updating them, but still no good. Now, question is How do you automatically update ` aws credentials`  when it expires?

We are using  `aws-sdk` for uploading &  viewing attachments such as images and pdf's on our web app. And we are also using this package on our mobile app too for uploading and viewing attachments as well.

**Comment by ashikjs** [comment]
@Selenophilia, we are facing same issue, after expired session we can't refresh token by `.get(), .refresh()` etc. event all of those method not working or executed !

@ajredniwja Can you help me by providing how those method work or how i can renew my token or session ?
